src/gui/torrentmodel.cpp

```diff
@@ -314,47 +298,53 @@ void TorrentModel::handleTorrentsUpdated()
 
 // Static functions
 
-QIcon getIconByState(BitTorrent::TorrentState state)
+static QString getIconId(BitTorrent::TorrentState state)
 {
     switch (state) {
     case BitTorrent::TorrentState::Downloading:
     case BitTorrent::TorrentState::ForcedDownloading:
     case BitTorrent::TorrentState::DownloadingMetadata:
-        return getDownloadingIcon();
+        return "downloading";
     case BitTorrent::TorrentState::Allocating:
     case BitTorrent::TorrentState::StalledDownloading:
-        return getStalledDownloadingIcon();
+        return "stalledDL";
     case BitTorrent::TorrentState::StalledUploading:
-        return getStalledUploadingIcon();
+        return "stalledUP";
     case BitTorrent::TorrentState::Uploading:
     case BitTorrent::TorrentState::ForcedUploading:
-        return getUploadingIcon();
+        return "uploading";
     case BitTorrent::TorrentState::PausedDownloading:
-        return getPausedIcon();
+        return "paused";
     case BitTorrent::TorrentState::PausedUploading:
-        return getCompletedIcon();
+        return "completed";
     case BitTorrent::TorrentState::QueuedDownloading:
     case BitTorrent::TorrentState::QueuedUploading:
-        return getQueuedIcon();
+        return "queued";
     case BitTorrent::TorrentState::CheckingDownloading:
     case BitTorrent::TorrentState::CheckingUploading:
     case BitTorrent::TorrentState::QueuedForChecking:
     case BitTorrent::TorrentState::CheckingResumeData:
-        return getCheckingIcon();
+        return "checking";
     case BitTorrent::TorrentState::Unknown:
     case BitTorrent::TorrentState::MissingFiles:
     case BitTorrent::TorrentState::Error:
-        return getErrorIcon();
+        return "error";
     default:
         Q_ASSERT(false);
-        return getErrorIcon();
+        return "error";
     }
 }
 
-QColor getColorByState(BitTorrent::TorrentState state)
+QIcon TorrentModel::getIconByState(BitTorrent::TorrentState state, const QString &overrideIconId)
+{
+    const QString &iconId(overrideIconId.isEmpty() ? getIconId(state) : overrideIconId);
```

Comment:
I did that initially but found it to be too complicated. But I thought I'd removed the `&overrideIcon` here as I don't think I use it anymore ... let me check.